sda/protocol.go

```diff
@@ -59,6 +59,10 @@ func ProtocolRegisterName(name string, protocol NewProtocol) ProtocolID {
 		protocols = make(map[ProtocolID]NewProtocol)
 		protocolNames = make(map[ProtocolID]string)
 	}
+	if _, exists := protocolNames[u]; exists {
+		log.Warn("Protocol", u, "already exists - not overwriting")
```

Comment:
In test/simulation-mode multiple hosts are handled by a cothority. If a protocol is registered more than once, it is mostly due to an error and will not have the desired effect.

My use-case was registering a bftcosi-protocol in the service, but giving it an individual verification-method for each node. This is not possible when having multiple nodes/hosts in the same cothority-binary, but only confusing the user (in that case, me). So give out a warning so the user knows he's doing something that the system won't support.